### PR TITLE
Add per-strip translucent polygons sort option

### PR DESCRIPTION
**Pull request by flyinghead**
User can now select per-triangle (default), per-strip and per-pixel (GL 4.3 only)
The per-strip mode should help Shenmue users who can't use GL 4.3

 Use FB_R_SOF2 only to detect direct FB writes (Virtua Fighter 3tb flashing FMV)

Other reset-related fixes

### CODE REVIEW

**Comment by skmp** [comment]
This pull request **fixes 1 alert** when merging c050ee8f74b735212a8f4b7c51d45ffc4c357a1f into a6b773a23e4167f953108e90711d5eeceed37f11 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-1b654f66fba9f96884a99f5d97c16d17a49ff315)

**fixed alerts:**

* 1 for Non\-virtual destructor in base class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **fixes 1 alert** when merging ee7dcf40e9493bb262e6ebde3dc248df18591358 into a6b773a23e4167f953108e90711d5eeceed37f11 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-cca28a2151ea1e41d64416e53c5b9eb6cc6c08e7)

**fixed alerts:**

* 1 for Non\-virtual destructor in base class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **fixes 1 alert** when merging 5a7421b7facb5fb6dbeeb3e81f12b49bd53a837c into a6b773a23e4167f953108e90711d5eeceed37f11 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-87fcacf6fe5eda34a58ea9c2a7e73babff039a3f)

**fixed alerts:**

* 1 for Non\-virtual destructor in base class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **fixes 1 alert** when merging 15087c7b9354e649971ae3ad0de7d4c76f29d341 into a6b773a23e4167f953108e90711d5eeceed37f11 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-aa106520dbaf62e5461940910db0928661491cc1)

**fixed alerts:**

* 1 for Non\-virtual destructor in base class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **fixes 1 alert** when merging 778bf951127d1839423b6e131d1a0d5494fa4b6c into a6b773a23e4167f953108e90711d5eeceed37f11 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-caf58be39be80ac15823baa69f7902b8467138a3)

**fixed alerts:**

* 1 for Non\-virtual destructor in base class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **fixes 1 alert** when merging 488462e3044438be64f3cc5090999564b058d1af into a6b773a23e4167f953108e90711d5eeceed37f11 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-e6389753c70616556e82bfce25656982381b93c8)

**fixed alerts:**

* 1 for Non\-virtual destructor in base class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: appveyor.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ before_build:
   - cmd: >-
       if "%CONFIGURATION%"=="RelWithDebInfo" (set BUILD_PATH=build)
       
+      if "%CONFIGURATION%"=="RelWithDebInfo" ("C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat" x64)
```
Comment (skmp) [inline appveyor.yml]: Is the hardcoded path here intentional?

File: core/hw/aica/aica_if.cpp [hunk 2]
```diff
@@ -160,6 +160,8 @@ void aica_Init()
 void aica_Reset(bool Manual)
 {
 	aica_Init();
+	VREG = 0;
+	ARMRST = 0;
 }
```
Comment (skmp) [inline core/hw/aica/aica_if.cpp]: 👌

File: core/hw/gdrom/gdromv3.cpp [hunk 3]
```diff
@@ -322,6 +316,13 @@ u32 GetFAD(u8* data, bool msf)
 void libCore_gdrom_disc_change()
 {
 	gd_setdisc();
+	read_params = { 0 };
```
Comment (skmp) [inline core/hw/gdrom/gdromv3.cpp]: I‘d add a comment here, or move to a `reset()` function

File: core/hw/pvr/Renderer_if.cpp [hunk 4]
```diff
@@ -558,10 +556,8 @@ void rend_vblank()
 void check_framebuffer_write()
 {
 	u32 fb_size = (FB_R_SIZE.fb_y_size + 1) * (FB_R_SIZE.fb_x_size + FB_R_SIZE.fb_modulus) * 4;
-	fb1_watch_addr_start = FB_R_SOF1 & VRAM_MASK;
```
Comment (skmp) [inline core/hw/pvr/Renderer_if.cpp]: Why not use page locks for this? Ehy only watch one field?

File: core/hw/gdrom/gdromv3.cpp [hunk 5]
```diff
@@ -322,6 +316,13 @@ u32 GetFAD(u8* data, bool msf)
...
+	set_mode_offset = 0;
+	packet_cmd = { 0 };
+	read_buff = { 0 };
```
Comment (davediff) [inline core/hw/gdrom/gdromv3.cpp]:
```suggestion
	read_buff.cache_index = 0;
	read_buff.cache_size = 0;
	memset(read_buff.cache, 0, 2352 * 8192);
```
Any issue with resetting the struct variables explicitly like this? It fixes the msvc heap size issue for me. Not sure I fully understand why.

**Pull request closed without merging**